source/_components/sensor.magicseaweed.markdown

```diff
@@ -0,0 +1,95 @@
+---
+layout: page
+title: "Magicseaweed Sensor"
+description: "How to integrate Magicseaweed within Home Assistant."
+date: 2018-06-24 21:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: magicseaweed.png
+ha_category: Sensor
+featured: false
+ha_release: "0.75"
+ha_iot_class: "Cloud Polling"
+---
+
+The 'magicseaweed' platform uses the [Magicseaweed Forecast API](https://magicseaweed.com/developer/forecast-api) as a source for surf forecasting data for the surf spots of your choice.
+
+You need an API key which is free but requires [registration](https://magicseaweed.com/developer/sign-up). Magicseaweed is limiting users of the API to 2 spots for the free plan.
+
+To add Magicseaweed forecasts to your installation, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+sensor:
+  - platform: magicseaweed
+    api_key: YOUR_API_KEY
+    spot_id: 1092
+    monitored_conditions:
+      - max_breaking_swell
+```
+
+{% configuration %}
+api_key:
+  description: The API key to access the service.
+  required: true
+  type: string
+name:
+  description: Spot nickname for the sensors.
+  required: false
+  default: MSW.
+  type: string
+hour:
+  description: List of hours you would like to receive data for.
+  required: false
+  default: Defaults to current forecast.
+  type: list
+  keys:
+    3AM:
+      description: Display forecast for 3AM.
+    6AM:
+      description: Display forecast for 6AM.
+    9AM:
+      description: Display forecast for 9AM.
+    12PM:
+      description: Display forecast for 12PM.
+    3PM:
+      description: Display forecast for 3PM.
+    6PM:
+      description: Display forecast for 6PM.
+    9PM:
+      description: Display forecast for 9PM.
+    12AM:
+      description: Display forecast for 12AM.
+spot_id:
+  description: Surf spot ID to monitor surf forecast of.Details for getting spot id available at [Magicseaweed](https://magicseaweed.com/developer/forecast-api)
+  required: true
+  type: string
+monitored_conditions:
+  description: Conditions to display in the frontend.
```

Comment:
This isn't dependent on the frontend per se, but decides what entities to create, ie what data to make available.